pkgs/misc/cups/drivers/brgenml1lpr/default.nix

```diff
@@ -0,0 +1,93 @@
+{ stdenv, fetchurl, cups, perl, glibc, ghostscript, which, makeWrapper}:
+
+/*
+    [Setup instructions](http://support.brother.com/g/s/id/linux/en/instruction_prn1a.html).
+
+    URI example
+     ~  `lpd://BRW0080927AFBCE/binary_p1`
+
+    Logging
+    -------
+    
+    `/tmp/br_lpdfilter_ml1.log` when `$ENV{LPD_DEBUG} > 0` in `filter_BrGenML1`
+    which is activated automatically when `DEBUG > 0` in `brother_lpdwrapper_BrGenML1`
+    from the cups wrapper.
+
+    Issues
+    ------
+
+     -  filter_BrGenML1 ln 196 `my $GHOST_SCRIPT=`which gs`;`
+      
+        `GHOST_SCRIPT` is empty resulting in an empty `/tmp/br_lpdfilter_ml1_gsout.dat` file.
+        See `/tmp/br_lpdfilter_ml1.log` for the executed command.
+
+    Notes
+    -----
+
+     -  The `setupPrintcap` has totally no use in our context.
+*/
+
+let myPatchElf = (file: 
+  if stdenv.system == "i686-linux" then ''
+    patchelf --set-interpreter ${stdenv.glibc}/lib/ld-linux.so.2 ${file}
+  '' else if stdenv.system == "x86_64-linux" then ''
+    patchelf --set-interpreter ${stdenv.glibc}/lib/ld-linux-x86-64.so.2 ${file}
+  '' else "");
+in
+stdenv.mkDerivation rec {
+
+  name = "brgenml1lpr-3.1.0-1";
+  src = fetchurl {
+    url = "http://download.brother.com/welcome/dlf101123/${name}.i386.deb";
+    sha256 = "0zdvjnrjrz9sba0k525linxp55lr4cyivfhqbkq1c11br2nvy09f";
+  };
+
+  unpackPhase = ''
+    ar x $src
+    tar xfvz data.tar.gz
+  '';
+
+  buildInputs = [ cups perl glibc ghostscript which makeWrapper];
```

Comment:
done for `makeWrapper`. `perl` is however needed at runtime as interpreter of `filter_BrGenML1` (introduced at automatic interpreter patching step).